ContextDesktop/src/main/java/com/github/migi_1/Context/enemy/EnemyFactory.java

```diff
@@ -1,20 +1,76 @@
  * @author Damian
  *
  */
-public class EnemyFactory extends AbstractObstacleFactory {
-
-    @Override
-    public Obstacle produce() {
-        // TODO Auto-generated method stub
-        return null;
+public class EnemyFactory {
+    
+    private BoundingBox levelPieceBoundingBox;
+    private float levelPieceLength;
+    private float levelPieceWidth;
+    private ArrayList<Carrier> carriers;
+    
+    /**
+     * Constructor of the EnemyFactory.
+     * @param levelPieceBoundingBox the boundingbox of the levelPiece.
+     * @param levelPieceLength Length of the levelPiece.
+     * @param levelPieceWidth Width of the levelPiece.
+     * @param carriers the carriers on which the spawning will be based.
+     */
+    public EnemyFactory(BoundingBox levelPieceBoundingBox, float levelPieceLength,
+            float levelPieceWidth, ArrayList<Carrier> carriers) {
+        this.levelPieceBoundingBox = levelPieceBoundingBox;
+        this.levelPieceLength = levelPieceLength;
+        this.levelPieceWidth =  levelPieceWidth;
+        this.carriers = carriers;
+    }
+    
+    /**
+     * Creates an enemy at the left side of the path, 2 levelPieces ahead of the commander.
+     * @param currentLevelPiece to calculate where to spawn the enemy
+     * @return the created enemy.
+     */
+    public Enemy createEnemy1(double currentLevelPiece) {
+        Enemy enemy =  new Enemy(new Vector3f(-(((int) currentLevelPiece + 2) * levelPieceLength),
+                carriers.get(0).getModel().getLocalTranslation().y,
+                levelPieceWidth + (levelPieceBoundingBox.getZExtent() / 2)),
+                carriers); 
+        enemy.getModel().rotate(0, (float) Math.PI, 0);
+        return enemy;
+    }
+    
+    /**
+     * Creates an enemy at the left side of the path, 2 levelPieces ahead of the commander.
+     * @param currentLevelPiece to calculate where to spawn the enemy
+     * @return the created enemy.
+     */
+    public Enemy createEnemy2(double currentLevelPiece) {
+        Enemy enemy =  new Enemy(new Vector3f(-(((int) currentLevelPiece + 2) * levelPieceLength)
+                + levelPieceLength * 1 / 2, carriers.get(0).getModel().getLocalTranslation().y,
+                levelPieceWidth + (levelPieceBoundingBox.getZExtent() / 2)),
+                carriers);        
+        enemy.getModel().rotate(0, (float) Math.PI, 0);
+        return enemy;
+    }
+    
+    /**
+     * Creates an enemy at the left side of the path, 2.5 levelPieces ahead of the commander.
+     * @param currentLevelPiece to calculate where to spawn the enemy
+     * @return the created enemy.
+     */
+    public Enemy createEnemy3(double currentLevelPiece) {
+        return new Enemy(new Vector3f(-(((int) currentLevelPiece + 2) * levelPieceLength)
+                + levelPieceLength * 1 / 2, carriers.get(0).getModel().getLocalTranslation().y,
+                levelPieceWidth - (levelPieceBoundingBox.getZExtent() / 2)),
+                carriers); 
```

Comment:
Enemy creation is the same in the three methods above. Maybe you can combine the three methods or create a method specifically for enemy creation?